Replicator/ReplicatorOptions.hh

```diff
@@ -36,20 +39,24 @@ namespace litecore { namespace repl {
         PropertyDecryptor       propertyDecryptor       {nullptr};
         void*                   callbackContext         {nullptr};
         std::atomic<C4ReplicatorProgressLevel> progressLevel {kC4ReplProgressOverall};
-        bool                    collectionAware         {true};
-
+        mutable bool            collectionAware         {true};
+        mutable bool            isActive                {true};
+        mutable std::unordered_map<C4CollectionSpec, size_t> collectionSpecToIndex;
```

Comment:
Why does this need to be explicitly marked mutable?  Doing this should be an exceptional case (mutating inside of const function for some reason)

Reply:
See above. Once adopted by Replicator and its workers, it is designated as constant, and this map is is computed inside Replicator::Replicator(). Actually, I am still working on it, CBL-3406, I find I have access to the non-const options in Replicator's constructor, and can use that to avoid mutable. 
One style I mean to adhere to for this struct is to have direct access to the members as much as possible.